ui/src/app/services/error-handling.service.ts

```diff
@@ -0,0 +1,59 @@
+import {Injectable, NgZone} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+
+@Injectable()
+export class ErrorHandlingService {
+
+  // Expose "current user details" as an Observable
+  public fiveHundred: boolean;
+  public zero: boolean;
+
+  constructor(private zone: NgZone) {
+    this.fiveHundred = false;
+    this.zero = false;
+  }
+
+  public publishFiveHundred(): void {
```

Comment:
It looks like this doesn't really "publish" values, in the sense of providing them as observables or creating events -- consumers of this service just use the values (in a template) and Angular's state tracking is what takes care of publishing the values? If that's accurate, I'd prefer the more typical `setX` and `clearX`. Or, since the fields are public anyway (for reading), accessors may be superflouous.

Reply:
Agreed. That makes more sense. (That's left over from when I thought I'd need to use observables.)